src/CabanaPD_ForceModels.hpp

```diff
@@ -71,6 +72,45 @@ struct BaseTemperatureModel
     }
 };
 
+// This class stores temperature parameters needed for heat transfer, but not
+// the temperature itself (stored instead in the static temperature class
+// above).
+struct BaseDynamicTemperatureModel
+{
+    double delta;
+
+    double thermal_coeff;
+    double kappa;
+    double cp;
+    bool constant_conductivity;
+
+    BaseDynamicTemperatureModel( const double _delta, const double _kappa,
+                                 const double _cp,
+                                 const bool _constant_conductivity = true )
+    {
+        set_param( _delta, _kappa, _cp, _constant_conductivity );
+    }
+
+    void set_param( const double _delta, const double _kappa, const double _cp,
+                    const bool _constant_conductivity )
+    {
+        delta = _delta;
+        kappa = _kappa;
+        cp = _cp;
+        const double d3 = _delta * _delta * _delta;
+        thermal_coeff = 9.0 / 2.0 * _kappa / pi / d3;
+        constant_conductivity = _constant_conductivity;
+    }
+
+    KOKKOS_INLINE_FUNCTION double microconductivity_function( double r ) const
```

Comment:
Is this way preferred to simply compute thermal_coeff for both "constant" and "conical" micro-conductivity using, e.g., an if statement?

Reply:
I guess my confusion was that we first compute "thermal_coeff = 9.0 / 2.0 * _kappa / pi / d3" and then possibly modify it in microconductivity_function. The choice of thermal_coeff = 9.0 / 2.0 * _kappa / pi / d3 seems arbitrary.

I was wonder whether it would be more clear to simply have in microconductivity_function, something like:

const double d3 = _delta * _delta * _delta;
if ( constant_microconductivity )
            return 9.0 * _kappa / (2.0*pi) / d3;
        else
            return 18.0 * _kappa / pi / d3 * ( 1.0 - r / delta );

without defining "thermal_coeff"